### PR TITLE
Add golden test comparing to mlpack for minimum spanning tree

### PR DESCRIPTION
**Pull request by dalg24**
Implement suggestion in https://github.com/arborx/ArborX/pull/624#discussion_r803838312

### CODE REVIEW

**Comment by dalg24** [comment]
Retest this please

File: test/tstMinimumSpanningTreeGoldenTest.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
```
Comment (aprokop) [inline test/tstMinimumSpanningTreeGoldenTest.cpp:2]:
```suggestion
 * Copyright (c) 2017-2022 by the ArborX authors                            *
```

File: test/tstMinimumSpanningTreeGoldenTest.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+struct UndirectedEdge : ArborX::Details::WeightedEdge
+{
+private:
+  friend bool operator==(UndirectedEdge const &lhs, UndirectedEdge const &rhs)
+  {
+    auto const lhs_min = std::min(lhs.source, lhs.target);
+    auto const lhs_max = std::max(lhs.source, lhs.target);
+    auto const rhs_min = std::min(rhs.source, rhs.target);
+    auto const rhs_max = std::max(rhs.source, rhs.target);
+    return (lhs_min == rhs_min) && (lhs_max == rhs_max);
```
Comment (aprokop) [inline test/tstMinimumSpanningTreeGoldenTest.cpp:34]:
```suggestion
    return ((lhs.source == rhs.source && lhs.target == rhs.target) ||
            (lhs.source == rhs.target && lhs.target == rhs.source));
```

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -165,9 +165,20 @@ target_compile_definitions(ArborX_Clustering.exe PRIVATE BOOST_TEST_DYN_LINK)
 target_include_directories(ArborX_Clustering.exe PRIVATE ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SOURCE_DIR}/examples/dbscan)
 add_test(NAME ArborX_Clustering_Test COMMAND ./ArborX_Clustering.exe)
 
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/mst_golden_test_points.csv
+  ${CMAKE_CURRENT_BINARY_DIR}/mst_golden_test_points.csv
+  COPYONLY
+)
```
Comment (aprokop) [inline test/CMakeLists.txt:172]: I would add a comment on the origin of these files.

File: test/tstMinimumSpanningTreeGoldenTest.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    edges.emplace_back(ArborX::Details::WeightedEdge{
+        static_cast<int>(std::stof(*first++)),
+        static_cast<int>(std::stof(*first++)), std::stof(*first++)});
+    assert(first == last);
+  }
+  return edges;
+}
+} // namespace Test
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(minimum_spanning_tree_compare, DeviceType,
```
Comment (aprokop) [inline test/tstMinimumSpanningTreeGoldenTest.cpp:84]:
```suggestion
BOOST_AUTO_TEST_CASE_TEMPLATE(minimum_spanning_tree_compare_gold, DeviceType,
```

**Review by aprokop — DISMISSED** [review]
LGTM

**Review by aprokop — APPROVED** [review]
